src/svg-icon.jsx

```diff
@@ -112,12 +105,17 @@ const SvgIcon = React.createClass({
       ...other,
     } = this.props;
 
+    const {
+      baseTheme,
+      prepareStyles,
+    } = this.state.muiTheme;
+
     const offColor = color ? color :
       style && style.fill ? style.fill :
-      this.state.muiTheme.rawTheme.palette.textColor;
+      baseTheme.palette.textColor;
     const onColor = hoverColor ? hoverColor : offColor;
 
-    const mergedStyles = this.mergeStyles({
+    const mergedStyles = Object.assign({
```

Comment:
I agree with @alitaheri, we don't need this. Let's make this component simpler.
What's the point of providing `fill` when this has no impact?